tools/font-edit/twin-fedit.c

```diff
@@ -492,95 +497,102 @@ static void button(char_t *c, XButtonEvent *bev)
 
 static void play(char_t *c)
 {
-    XEvent ev;
-    char key_string[10];
-
-    XClearArea(dpy, win, 0, 0, 0, 0, True);
-    for (;;) {
-        XNextEvent(dpy, &ev);
-        switch (ev.type) {
-        case KeyPress:
-            if (XLookupString((XKeyEvent *) &ev, key_string, sizeof(key_string),
-                              0, 0) == 1) {
-                switch (key_string[0]) {
-                case 'q':
-                    return;
-                case 'c':
-                    XClearArea(dpy, ev.xkey.window, 0, 0, 0, 0, True);
+    /* Ensures that SDL only focuses on key events*/
+    SDL_StopTextInput();
+
+    SDL_Event event;
+    int quit_state = 0;
+
+    /* keep track of the selected spline */
+    cmd_t *spline = NULL;
+    while (!quit_state) {
+        while (SDL_PollEvent(&event)) {
+            switch (event.type) {
+            /* If SDL event is detected */
+            case SDL_QUIT:
+                /* Click the "X" on the top of screen to exit the program */
+                quit_state = 1;
+                exit_window = 1;
+                break;
+            case SDL_KEYDOWN:
+                /* If any key event is detected */
+                SDL_Keycode key_event = event.key.keysym.sym;
+                switch (key_event) {
+                case SDLK_q:
+                    /* Press "q" key to exit play()
+                     * This allows the user to display the next character.
+                     */
+                    quit_state = 1;
```

Comment:
It is quite ambiguous to name `quit_state` since the latter is not meant to quit the application.

Reply:
Resolve in https://github.com/sysprog21/mado/commit/7231d62d0106fb56594ddd7785efcc9ab9779b04